src/LibreCards.Core/CardRepository.cs

```diff
@@ -0,0 +1,66 @@
+﻿using LibreCards.Core.Entities;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+
+namespace LibreCards.Core
+{
+    public class CardRepository : ICardRepository
+    {
+        private readonly string _filePath = Path.Combine(
+            Directory.GetParent(Environment.CurrentDirectory)
+            .Parent
+            .Parent
+            .Parent
+            .FullName,
+            @"LibreCards.Core\default-cards.json");
+        private int _responseCardsIndex;
+
+        public IEnumerable<Card> DrawCards(int count = 1)
+        {
+            if (!File.Exists(_filePath))
+            {
+                throw new FileNotFoundException();
+            }
+
+            byte[] rawFileContent;
+            try
+            {
+                rawFileContent = File.ReadAllBytes(_filePath);
+            }
+            catch (Exception exception)
+            {
+                throw new Exception(exception.Message);
+            }
+
+            JsonFileStructure jsonInterpretation;
+            try
+            {
+                jsonInterpretation = JsonSerializer.Deserialize<JsonFileStructure>(rawFileContent);
+            }
+            catch (JsonException)
+            {
+                throw new JsonException("Cards collection could not be deserialized");
+            }
```

Comment:
I think it would be a bit more beneficial to separate some responsibilities here.

For example, I think we should abstract the System.IO calls into something like `IDataStorage` so that we can then mock it through our tests without the need to have an actual default-cards.json file included.

`IDataStorage` could even have `DefaultCards { get; }` to completely hide the JSON implementation.

Reply:
Ok, good idea, will do.

Reply:
Solved by last commit